src/common/bitstream.cpp

```diff
@@ -0,0 +1,392 @@
+#include <algorithm>
+#include "bitstream.h"
+#include "util.h"
+
+#define BITS_TO_BYTES(bits) (((bits) + 7) / 8)
+
+
+BitStream::BitStream(std::vector<uint8_t>& exisitingBuf) :
+    buf(exisitingBuf),
+    streamBitPos(0),
+    lastError(Error::NONE) {
+
+}
+
+size_t BitStream::getSizeBits() const {
+    return buf.size() * 8;
+}
+
+size_t BitStream::getRemainingBits() const {
+    size_t sizeBits = getSizeBits();
+    if(streamBitPos >= sizeBits) {
+        return 0;
+    }
+    else {
```

Comment:
Hmm, I don't think VS has an option for that specific case, although I'm wondering if it might not autoformat if you newline it manually. Hopefully we shouldn't have too many single-line conditionals that it would be a problem. That's the main thing I'm worried about actually - not so much the specifics of what particular style we actually use, but just that VS autoformatting will sprinkle all sorts of irrelevant little changes into the commit/PR diffs if we have different settings and make them harder to read.